tools/timemachine/internal/cmd/download.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/iterator"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+// const value
+var (
+	GcpTimeout  = time.Second * 60
+	ErrNotExist = errors.New("height does not exist")
+)
+
+// download represents the download command
+var download = &cobra.Command{
+	Use:   "download [height] [directoy]",
+	Short: "Download height block datas into directoy",
+	Args:  cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		height, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to convert input height: %s.", args[0])
+		}
+		if height == 0 {
+			return errors.New("input height cannot be 0")
+		}
+
+		var (
+			destDir = args[1]
+			wg      sync.WaitGroup
+		)
+		const (
+			_bucket    = "blockchain-golden"
+			_objPrefix = "fullsync/mainnet/"
+		)
+
+		heightDir := genHeightDir(height)
+		log.S().Infof("download the height's dir: %s", heightDir)
+
+		objs, err := listFiles(_bucket, _objPrefix+heightDir+"/", "/")
+		if err != nil {
+			return err
+		}
+		if len(objs) == 0 {
+			objs, err = listFiles(_bucket, _objPrefix+"latest"+"/", "/")
```

Comment:
return `latest` in `genHeightDir`, so don't need to check here

Reply:
fixed